modules/openxAnalyticsAdapter.js

```diff
@@ -364,94 +366,93 @@ function onSlotLoaded({ slot }) {
+  // Backwards compatibility for external documentation
+  if(adapterConfig.options.slotLoadWaitTime){
+    adapterConfig.options.payloadWaitTime = adapterConfig.options.slotLoadWaitTime;
   }
-});
-
-openxAdapter.originEnableAnalytics = openxAdapter.enableAnalytics;
 
-openxAdapter.enableAnalytics = function(adapterConfig) {
-  analyticsConfig = {...DEFAULT_ANALYTICS_CONFIG, ...adapterConfig.options};
-  analyticsConfig.testCode = getTestCode();
-  analyticsConfig.utmTagData = this.buildUtmTagData();
-  utils.logInfo('OpenX Analytics enabled with config', analyticsConfig);
+  if(isValidConfig(adapterConfig)){
+    analyticsConfig = {...DEFAULT_ANALYTICS_CONFIG, ...adapterConfig.options};
+    analyticsConfig.utmTagData = this.buildUtmTagData();
+    utils.logInfo('OpenX Analytics enabled with config', analyticsConfig);
+
+    if (analyticsConfig.testPipeline) {
+      openxAdapter.track = (args) => {
+        prebidAnalyticsEventHandlerV1(args);
+        prebidAnalyticsEventHandlerV2(args);
+      };
+
+      googletag.cmd.push(function() {
+        googletag.pubads().addEventListener(SLOT_LOADED, args => {
+          utils.logInfo('OX: SlotOnLoad event triggered');
+          onSlotLoaded(args);
+          onSlotLoadedV2(args);
+        });
+      });
 
-  if (analyticsConfig.testPipeline) {
-    // override track method with v2 handlers
-    openxAdapter.track = prebidAnalyticsEventHandler;
+    } else if (analyticsConfig.enableV2) {
+      // override track method with v2 handlers
+      openxAdapter.track = prebidAnalyticsEventHandlerV2;
 
-    googletag.cmd.push(function() {
-      googletag.pubads().addEventListener(SLOT_LOADED, args => {
-        openxAdapter.track({ eventType: SLOT_LOADED, args });
-        utils.logInfo('OX: SlotOnLoad event triggered');
+      googletag.cmd.push(function() {
+        googletag.pubads().addEventListener(SLOT_LOADED, args => {
+          openxAdapter.track({ eventType: SLOT_LOADED, args });
+          utils.logInfo('OX: SlotOnLoad event triggered');
+        });
       });
-    });
-  } else {
-    googletag.cmd.push(function() {
-      googletag.pubads().addEventListener(SLOT_LOADED, function(args) {
-        utils.logInfo('OX: SlotOnLoad event triggered');
-        onSlotLoaded(args);
+    } else {
+      openxAdapter.track = prebidAnalyticsEventHandlerV1;
+      googletag.cmd.push(function() {
+        googletag.pubads().addEventListener(SLOT_LOADED, function(args) {
+          utils.logInfo('OX: SlotOnLoad event triggered');
+          onSlotLoaded(args);
+        });
       });
-    });
+    }
+
+    openxAdapter.originEnableAnalytics(adapterConfig);
   }
 
-  openxAdapter.originEnableAnalytics(adapterConfig);
+  function isValidConfig({options: analyticsOptions}){
+    const fieldValidations = [
+      // tuple of property, type, required
+      ['publisherPlatformId', 'string', true],
+      ['publisherAccountId', 'number', true],
```

Comment:
If possible, can we make this as a string, since our openxBidAdapter requires the adunit as a string? 

Because if we take those users into account, then they will get the 2 confused and wonder why 1 is a number and why 1 is a string.

See http://prebid.org/dev-docs/bidders/openx.html > unit

Reply:
It's currently a number and a `long` in the backend.